\chapter{Materials \& Methods}

%This section should include a recipe of what you did (explain what you have done so if someone wants to reproduce the experiment, they can).  A flow chart is typically helpful.  Also, make sure to define all software that you used including version numbers and OS.  Should also include a description of statistical methods used (if any).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Preliminary Study}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
The transcriptomic data used as the basis of this dissertation were collected by Dr Vassallo Gatt in her doctoral study \citep{Gatt2016}, originating from samples of the HL-60 cell line subjected to different treatments. Three samples were incubated and treated with a phenol for varying lengths of time (1, 6 and 12 hours), while a fourth sample served as the negative control, using the growth medium RPMI 1640 as the 'treatment'. This section is a summary of the laboratory procedure utilised by Dr Vassallo Gatt, and is intended to give context to the data.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Phenol extraction}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Phenolic compounds were isolated from Maltese extra-virgin olive oil by \ac{LLE}(REF method), followed by separation of fractions using preparative-scale \ac{HPLC}. \ac{LLE} transferred the water-soluble compounds (some of which phenols) from their organic solvent to an aqueous one. The heavier aqueous solution was extracted using a separating funnel while the raffinate was discarded. \ac{HPLC} was used to separate the components of the remaining solution based on their differing chemical interactions with an adsorbent column. As the solution was pumped through the column, the targeted compound flowed at a different rate from the other compounds, leading to its isolation. The compound was identified as a phenol but further analysis is required to determine its precise chemical composition. 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Cell Culturing and RNA extraction}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%6-well plate or 96-well?
HL-60 cells were mixed with the phenol and the mixture was used to seed three wells out of a 6-well plate. The control was seeded similarly, but with RPMI 1640 used instead of the phenol. Samples were incubated on RPMI 1640 for their stipulated time period, after which the treated cells began showing characteristic morphological signs of differentiation such as the presence of lobed nuclei, vacuoles, and a decreased nucleus:cytoplasm ratio.

The samples were frozen, then thawed on ice and subjected to RNA extraction according to the RNeasyÂ® Mini kit \citep{RNeasy}. This involved the separation of the mixture into two partitions: an organic phase containing DNA and protein, and an aqueous phase containing the RNA, induced by the addition of chloroform. The aqueous phase was separated into a separate microcentrifuge tube to which ethanol was added for precipitation of nucleic acids. The mixture was subjected to multiple cycles of centrifugation. Between centrifugation cycles the flow-though was discarded and lysis buffer was added to remove silica-bound proteins, carbohydrates, fatty acids and any traces of salts. A final centrifugation was performed to elute the RNA in RNase-free water.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Determination of RNA Quality and Sequencing}%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Elution/aliquots?
%aliquot = a portion of a larger whole
The RNA was analysed using a spectrophotometer which confirmed that the concentrations and A260/280 values were of acceptable quality. The total RNA extracted was then shipped to the \ac{EMBL} in Heidelberg, Germany where the RNA integrity was analysed by gel analysis and then sequenced using an Illumina HiSeq 2000 Sequencing System. The steps followed were typical of the Illumina \ac{NGS} workflow \citep{HiSeq2000}, consisting of library preparation, cluster generation, sequencing by synthesis, sequence identification, demultiplexing of the data and the assignment of Phred (Q) scores to each base call.

The transcriptomic data was received in the form of four FASTQ \citep{FASTQ} files, one per experimental time point. They were composed of 51  \ac{bp} single-ended reads, with 30x coverage, and were Sanger/Illumina 1.9 encoded (which uses the ASCII character corresponding to the Phred score, and adds '33' to it). These served as the starting point for our RNAseq pipeline.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{RNA-seq Pipeline}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\iffalse 
Tool versions:
MultiQC v1.11
STAR 2.7.9a
RSEM v1.3.3
EdgeR
Other R packages?
\fi

Bulk RNA-seq analysis is performed in a number of steps, each requiring one or more different tools. The data 'flows' through these tools, which are constituents of the pipeline. Choosing the tools was a non-trivial task, and the process is explained further in Chapter \ref{Assessing the Quality of the Raw FASTQ files}. The pipeline for this analysis is represented as the flowchart in \autoref{fig:Dissertation_pipeline}.
Data was stored and processed until the Quantification stage on a high performance computer, managed by the University of Malta with the following specifications: 56-core Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz with 125GB RAM, running the Ubuntu v18.04.5 operating system. Read count data was then transferred to a smaller computer also managed by the University of Malta, running a VirtualBox v6.1.32 \citep{virtualbox} virtual environment, with the following partitioned resources: Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz with 4GB RAM, running Ubuntu v20.04.4.

\begin{figure}[!ht]
    \centering
    \includegraphics[width=1\textwidth]{Dissertation_pipeline}
    \caption[General overview of the RNA-seq pipeline.]{General overview of the RNA-seq pipeline. Created with \href{https://biorender.com/}{BioRender.com}.} 
    \label{fig:Dissertation_pipeline}
\end{figure}

\clearpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Quality Control}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The first step of any sequencing pipeline should be the assessment of the quality of the raw data files. Sequencing information of poor quality is identified if present, and if necessary, truncated to mitigate inaccuracies in the downstream pipeline. To assess the quality of the FASTQ files, they were ported to FastQC v0.11.9 \citep{andrews2019} using eight threads (\autoref{lst:FastQC}). This provided summary information on, \textit{inter alia}, the Phred scores, GC content, N content, sequence length distribution, sequence duplication levels, overrepresented sequences and Kmer content of the sequences. FastQC rates each of these modules using a green check-mark signifying that it 'passed' QC, a yellow exclamation mark 'warning', or a red cross 'failed'. 

Two modules failed consistently throughout the four samples received: 'Sequence Duplication Levels' and 'Per base sequence content'. This is normal and expected for RNA data, and is explained in further detail in \autoref{Assessing the Quality of the Raw FASTQ files}. Traces (<0.5\%) of TruSeq adapters were detected in the FASTQ files, which is corroborated by the sequencer's manual \citep{HiSeq2000} stating that it makes use of the 'TruSeq family of reagents'. 

%I should explain FASTQC options a bit more

\begin{lstlisting}[language=bash, caption=FastQC command, label={lst:FastQC}]
# FastQC accepts multiple input files, so we can use wildcards
fastqc \
 -t 8 \
 -o ./raw_FastQC_out \
*.fq 
\end{lstlisting}

Next, Fastq Screen v0.14.0 \citep{wingett2018fastq} was used to check for RNA contaminants from other common sources, by comparing the reads against a set of sequence databases (\autoref{lst:FastqScreen}). Perl, an aligner (Bowtie \citep{bowtie}, Bowtie2 \citep{bowtie2} or BWA \citep{bwa}) and a Linux-based operating system are required. Bowtie2 was used to align the sample reads to the reads in the contaminant database.

\begin{lstlisting}[language=bash, caption=FastqScreen command, label={lst:FastqScreen}]
# FastQScreen also accepts multiple file inputs
fastq_screen \
--aligner bowtie2 \
--conf ./fastq_screen.conf \
 *.fq 
\end{lstlisting}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Preprocessing}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Poor quality reads lead to poor downstream sequence analysis. Thus, it is common practice in RNA-seq to trim the undesired regions. Cutadapt \citep{martin2011cutadapt} was used to trim the previously detected TruSeq adapter sequences and short reads, using a threshold of <45\ac{bp}. Between 1.3 and 1.6\% of the \ac{bp} were trimmed across the four samples. Cutadapt was accessed through the wrapper script Trim Galore! v0.6.7 \citep{trimgalore} which instantly redirects the trimmed data back to FastQC to assess the improvement in quality, if any.

\begin{lstlisting}[caption=Trim Galore! trimming]
trim_galore \
--phred33 \
--fastqc \
-a "GATCGGAAGAGCACACGTCTGAACTCCAGTCAC" \
--length 45 \
-o trim_galore_output \
--fastqc_args "-o trimmed_FastQC_out -t 8" \
--cores 4 \
*.fq 
\end{lstlisting}

The trimmed files were further filtered using Prinseq++ \citep{prinseq++} which removed ambiguous reads containing >7 N's and sequences with a DUST score\footnote{A value between 0 and 1 generated by the DUST algorithm which is a measure of sequence complexity. Here, its purpose is to mask low-complexity regions.} of <0.1.

\begin{lstlisting}[caption=Prinseq++ filtering]
samples=(control, 1hour, 6hour, 12hour)
mkdir prinseq_out

for sample in ${samples[@]};
do
        prinseq++ \
        -fastq ./trim_galore_output/${sample}_trimmed.fq \
        -out_name ./prinseq_out/${sample}_filtered.fq \
        -out_bad ./prinseq_out/${sample}_bad.fq \
        -ns_max_n 7 \
        -lc_dust=0.1 > ./prinseq_out/control_prinseq_report.txt
done
\end{lstlisting}

\subsection{Read alignment, Quantification and Normalisation}
The trimmed and filtered FASTQ files were aligned to the GRCh38.p13 reference genome \citep{ref} using the \ac{STAR} aligner \citep{Dobin2013}. \ac{STAR} was called through RSEM \citep{li2011rsem}, which after alignment estimates gene and isoform expression levels, and normalises these counts. RSEM must be accessed through a 64 bit Linux or Mac OS command-line, and must have C++, Perl and R installed as dependencies.

Reference transcripts from the reference genome were first generated through the \texttt{rsem-prepare-reference} program, with the help of its respective GTF file (\autoref{lst:RSEM_ref}). 

% .genes.results (one row per gene) which has the headers: gene_id transcript_id(s)        length  effective_length        expected_count  TPM     FPKM
% .isoforms.results (one row per transcript) which has the headers: transcript_id   gene_id length  effective_length        expected_count  TPM     FPKM    IsoPct

\begin{lstlisting}[caption=RSEM reference generation, label={lst:RSEM_ref}]
mkdir RSEM
mkdir RSEM/reference
rsem-prepare-reference --gtf mm9.gtf \
                       --star \
                       --star-sjdboverhang 50 \
                       --num-threads 8 \
                       --gtf Homo_sapiens.GRCh38.104.gtf \
                       Homo_sapiens.GRCh38.dna.primary_assembly.fa \
                       ./RSEM/reference/GRCh38
\end{lstlisting}

The \texttt{--star-sjdboverhang} is a parameter passed to \ac{STAR} which sets the maximum possible overhang for the reads. It should be equal to the read length minus one. In our case, the read length is 51, thus a value of 50 is used. The final variable is a prefix to the output files. This may be used as the output path for the generated reference files.

To calculate expression values, the \texttt{rsem-calculate-expression} command is used. As before, the final parameter is a prefix to the output files, and the parameter before that is the path to the reference files created in the previous command. RSEM first uses \ac{STAR} to align the filtered sample reads to the generated reference reads, creating a BAM file sorted by its coordinates. It outputs two quantification files per sample, with differing suffices: one ending in '.genes.results' and another '.isoform.results'. These are both tab-delimited text files with similar headers, with the former dedicating a row for each gene and the latter dedicating a row for each transcript. The isoform-centric file thus contains more data, which are unnecessary for our objectives. Of the columns in the gene-centric file, two will be used in the subsequent step: 'gene\textunderscore id' which holds the Ensembl gene ID, and 'TPM' which holds the normalised read count for the respective gene.

\begin{lstlisting}[caption=RSEM expression command, label={lst:RSEM_exp}]
mkdir RSEM/expression
samples=(control, 1hour, 6hour, 12hour)

for sample in ${samples[@]};
do
        rsem-calculate-expression \
        --num-threads 16 \
        --star \
        --sort-bam-by-coordinate \
        ./prinseq_out/${sample}*good_out.fq \
        ./RSEM/reference/GRCh38 \
        ./RSEM/expression/${sample}

done
\end{lstlisting}

\subsection{Reassessing the Quality}

The produced files from FastQC, FastQScreen, Trim Galore! and RSEM, were funnelled into MultiQC v1.11 \citep{multiqc} which summarises their output in the form of an HTML report. At the time of writing, the latest version of MultiQC did not support Prinseq++ and its log files were inspected manually instead.

\begin{lstlisting}[language=bash, caption=MultiQC command]
multiqc \
-o ./multiqc_out \
./trim_galore_output/*txt \
./fastqc/FastQC_out/. \
./FastQ_Screen_out/raw/. \
./RSEM/expression/.
\end{lstlisting}

\subsection{Differential Gene Expression}
\emph{NOTE: should I include code snippets here too?}
\emph{I should probably explain certain params and why i chose them in more depth. Or should this go in the discussion?}

We have chosen edgeR v3.14 \citep{edger} as the package to help identify differentially expressed genes, which supports datasets that lack replicates such as our own. The instructions on the vignette were followed, and are summarised as follows. 
%NOTE: raw read counts not normalised.

The four gene read count files generated through RSEM were imported into R v3.6.3 \citep{R} and had their gene IDs (1st column) and normalised read counts (6th column) compiled into a single \texttt{DGEList} object which consists of multiple data-frames. Genes with low counts (<10) are filtered out using \texttt{filterByExpr} and annotated using the AnnotationDbi package v3.14 \citep{annotationdbi} in conjunction with the org.Hs.eg.db v3.10.0 \citep{org.Hs.eg.db} human annotation database. This added the Entrez ID, gene symbol, associated GO terms and KEGG pathways to each gene. Certain annotations do not interact well with other parts of the downstream analysis as a result of one:many mapping however, so these were added in a separate, staggered steps.

EdgeR has multiple methods to determine which genes are differentially expressed, but the classic approach uwing the  \texttt{exactTest} function was deemed as most appropriate. This is a pairwise test which compares the means between two groups of counts with a negative-binomial distribution. The test was performed a total of three times, where each treated sample was compared to the control. 

\subsubsection{Calculating Dispersion}
An \texttt{exactTest} function requires the dispersion of our data as input, however calculating this is not mathematically possible due to our lack of replicates. In such cases the vignette suggests that estimating the dispersion based on the experimental conditions is more scientifically sound than assuming no variation. The \ac{BCV} is equal to the square root of the dispersion, and the vignette suggests a few values for the \ac{BCV} according to the experimental conditions, such as '0.1 for data on genetically identical model organisms'. While the data are human in origin, not a model organism, they originate from the same genetically identical cell line, thus 0.1 was deemed an appropriate estimation for the \ac{BCV}. The estimated dispersion value used for \texttt{exactTest} was equal to the BCV\textsuperscript{2}, or 0.01. This returns a \texttt{DGEExact} object for each pairwise comparison, containing the logFC, logCPM, p-values and annotations for each gene. 

\subsubsection{Cutoffs and Adjusting p-values}
To account for multiple testing, which is an inherent risk of RNA-seq due to its vast amounts of comparisons, p-values were adjusted using the Benjamini \& Hochberg method, also known by its alias, the \ac{FDR}. This value is, in short, the proportion of false positives one might expect to get from a test. Each of the \texttt{DGEExact} objects was transformed in this way using the \texttt{topTags} function. Differential expression was ranked by the \ac{FDR} and we filtered out any genes with an \ac{FDR} < 0.05. Filtering on the logFC was then performed using a custom script with a threshold of 1.5.  
%adjusting for multiple testing

\subsection{Functional analysis}



% EdgeR has three ways of judging differential expression but exactTest was deemed as the best fit

%Creating a DGEList which can be manipulated like any other list in R
% The main components of an DGEList object are a matrix counts containing the integer counts,
% a data.frame samples containing information about the samples or libraries, and a optional
% data.frame genes containing annotation for the genes or genomic features


%Summary of  https://www.science.smith.edu/cmbs/wp-content/uploads/sites/36/2015/09/P-and-q-values-in-RNA-Seq.pdf:
%		FDR makes use of Q-values which usually result in a smaller number of false positives, although this is not always the case:
%		A p-value of 0.05 implies that we are willing to accept that 5% of all tests will be false positives. 
%		An FDR-adjusted p-value (aka a q-value) of 0.05 implies that we are willing to accept that 5% of the tests found to be statistically significant (e.g. by p-value) will be false positives

%'Many traditional techniques such as the Bonferroni correction are too conservative in the sense that while they reduce the number of false positives, they also reduce the number of true discoveries.': https://www.nonlinear.com/support/progenesis/comet/faq/v2.0/pq-values.aspx

%EdgeR manual:
%A commonly used approach is to conduct DE tests, apply a fold-change cut-off and then rank all the genes above that fold-change threshold by p-value. In some other cases genes are first chosen according to a p-value cut-off and then sorted by their fold-changes.


%\subsubsection{Gene Set Enrichment Analysis and Pathway Analysis}
%\ac{GO}
%\ac{KEGG}
% https://www.biostars.org/p/462980/
% pathways
% library(org.Hs.eg.db)
% library(AnnotationDbi) 
% library(pathview)
% HOUSEKEEPING GENES for sanity checks 
%https://www.genomics-online.com/resources/16/5049/housekeeping-genes/

\section{Summary}
Empty for now.