---
title: "edgeR"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(edgeR)
library(dplyr)
```
https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf

A typical edgeR analysis might look like the following. Here we assume there 
are four RNASeq libraries in two groups, and the counts are stored in a 
tab-delimited text file, with genesymbols in a column called Symbol.

Files need to contain two columns, one for the counts and one for a gene identifier.
```{r}
files <- list("1hour.genes.results", "12hour.genes.results", "6hour.genes.results", "control.genes.results")
path <- "/home/bioinf/Desktop/RNAseq/RSEM_out/genes/"
```

Creating a DGEList which can be manipulated like any other list in R
Takes normalised counts as input
```{r}
group <- factor(c(1,2,3,4))
y <- readDGE(files, path=path, columns=c(1,6), group=group, labels=NULL)
```

Filters genes with low counts 
Threshold of 5-10 is normally taken
```{r}
keep <- filterByExpr(y, group=group)
table(keep)
y <- y[keep,,keep.lib.sizes=FALSE]
```

Normalisation by TMM
```{r}
y <- calcNormFactors(y)
design <- model.matrix(~group)
```

Dispersion cannot be calculated so we take the BCV as 0.1 since we have no replicates
```{r}
bcv <- 0.1
et <- exactTest(y, dispersion=bcv^2)
```

Display top differentially expressed tags
logFC, the log-abundance ratio, i.e. fold change, for each tag in the two groups being compared
logCPM, the log-average concentration/abundance for each tag in the two groups being compared
PValue, exact p-value for differential expression using the NB model
FDR, the p-value adjusted for multiple testing as found using p.adjust using the method specified
```{r}classic_toptags <- topTags(et, n=20, adjust.method="BH", sort.by="logFC")$table```

To perform likelihood ratio tests:
```{r}
fit <- glmFit(y,design, dispersion=bcv^2)
lrt <- glmLRT(fit,coef=4)
lrt_toptags <- topTags(lrt, n=20, adjust.method="BH", sort.by="logFC")$table
```
A commonly used approach is to conduct DE tests, apply a fold-change cut-off 
and then rank all the genes above that fold-change threshold by p-value
The following is a rigorous statistical test for thresholded hypotheses.
it tests whether the log2-fold-change is greater than lfc in absolute value.
```{r}
tr <- glmTreat(fit, coef=4, lfc=3)
tr_toptags <- topTags(tr)$table
```
Read normalized counts from the standard input.
```{r}
library(gplots)
data = read.table("stdin", header = T, sep = "\t", as.is = TRUE)
```
The gene names in the first column.
```{r}
gene = row.names(tr$table)
```

Load the data from the second column on.
```{r}
vals = as.matrix(tr$table[, 2 : ncol(tr$table)])
```

Adds a little noise to each element to avoid the
clustering function fail on zero variance datalines.
```{r}
vals = jitter(vals, factor = 1, amount = 0.00001)
```

Each row is normalized to a z-score
```{r}
zscore = NULL
for (i in 1 : nrow(vals)) {
  row = vals[i,]
  zrow = (row - mean(row)) / sd(row)
  zscore = rbind(zscore, zrow)
}
```
Add back gene names as row names.
```{r}
row.names(zscore) = gene
```

Turn it into a matrix for heatmap2.
```{r}
zscore = as.matrix(zscore)
```

Open the drawing device.
```{r}
pdf('|cat')
```

Set the color scheme.
```{r}
colors = colorRampPalette(c("green", "black", "red"), space = "rgb")(256)
```

Draw the heatmap.
```{r}
heatmap.2(zscore, col = colors, density.info = "none", trace = "none", margins = c(7, 7), lhei = c(1, 5))
```

Turn off the device.
```{r}
dev.off()
```




